### PR TITLE
[tools] php7 fix for fix_timepoint_date_problems.php

### PR DESCRIPTION
**Pull request by gluneau**
Bring the script to PHP 7 syntax.

### CODE REVIEW

File: tools/fix_timepoint_date_problems.php [hunk 1]
```diff
@@ -483,7 +483,7 @@ function diagnose($sessionID, $dateType=null, $newDate=null)
             fwrite(STDERR, "Age at $stage: $age [ $dateBirth $dateOfStage]\n");
 
             // create battery object
-            $battery =& new NDB_BVL_Battery();
+            $battery =& NDB_BVL_Battery();
```
Comment (johnsaigle) [inline tools/fix_timepoint_date_problems.php]: Should this be `=` instead of `=&` as above?

Comment (gluneau) [inline tools/fix_timepoint_date_problems.php]: We have it running like this on IBIS production already.

Comment (johnsaigle) [inline tools/fix_timepoint_date_problems.php]: Yeah I'm sure it doesn't produce errors or anything. You just made a change on line 254 that seems to update the same line of code and it's inconsistent with this one.

File: tools/fix_timepoint_date_problems.php [hunk 2]
```diff
@@ -454,11 +454,11 @@ function diagnose($sessionID, $dateType=null, $newDate=null)
 
     // check if the timepoint is started before attempting to make changes to it
     if ($timePoint->getCurrentStage() == 'Not Started' || empty($stageList['screening']['status'])) {
-        throw new LorisException("Error: Cannot diagnose the non-started timepoints!");
+        throw LorisException("Error: Cannot diagnose the non-started timepoints!");
```
Comment (ridz1208) [inline tools/fix_timepoint_date_problems.php]: PHP Fatal error:  Uncaught Error: Call to undefined function LorisException() in /var/www/loris/tools/fix_timepoint_date_problems.php:457

File: tools/fix_timepoint_date_problems.php [hunk 3]
```diff
@@ -237,12 +237,12 @@ function addInstrument($sessionID, $testName)
     // check the user $_ENV['USER']
     $user =& User::singleton(getenv('USER'));
     if($user->getUsername() == null) {
-        throw new LorisException("Error: Database user named " . getenv('USER') . " does not exist. Please create and then retry script\n");
+        throw new \LorisException("Error: Database user named " . getenv('USER') . " does not exist. Please create and then retry script\n");
```
Comment (driusan) [inline tools/fix_timepoint_date_problems.php]: is the `\` to get to the root namespace required or not?

If not, it shouldn't be here, if it is, all the other lines below require it too.

**Comment by leothomas** [comment]
@sruthymathew123 I've made some edits to avoid PHP notices and removed require_once statements which were breaking. Can you test to confirm that functionality is correct and error logs are clean of php errors/notices (LorisExceptions are fine)?

**Comment by sruthymathew123** [comment]
@leothomas @ridz1208 When I am using this script, it is just printing the full script on my screen. Nothing else happening.

File: tools/fix_timepoint_date_problems.php [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<?php
+?php
```
Comment (sruthymathew123) [inline tools/fix_timepoint_date_problems.php]: Why you changed this?

Comment (leothomas) [inline tools/fix_timepoint_date_problems.php]: I must have been in insert mode when I thought I wasn't. I'll edit and recommit. Sorry!

Comment (driusan) [inline tools/fix_timepoint_date_problems.php]: how do you delete a character by accidentally being in insert mode?

**Comment by sruthymathew123** [comment]
@leothomas 
PHP Notice:  Array to string conversion in /var/www/loris/php/libraries/Database.class.inc on line 688
PHP Notice:  Undefined index: ID in /var/www/loris/php/libraries/User.class.inc on line 72
PHP Notice:  Undefined index: Real_name in /var/www/loris/php/libraries/User.class.inc on line 86
PHP Notice:  Undefined variable: user_cid in /var/www/loris/php/libraries/User.class.inc on line 104
PHP Notice:  Undefined index: UserID in /var/www/loris/php/libraries/User.class.inc on line 217
PHP Fatal error:  Uncaught LorisException: Error: Database user named lorisadmin does not exist. Please create and then retry script
 in /var/www/loris/tools/fix_timepoint_date_problems.php:254
Stack trace:
#0 /var/www/loris/tools/fix_timepoint_date_problems.php(162): addInstrument('87', 'vineland_proban...')
#1 {main}
  thrown in /var/www/loris/tools/fix_timepoint_date_problems.php on line 254

**Comment by leothomas** [comment]
@sruthymathew123 Rida helped me with this issue, the script is looking for a user in your database, and since you haven't logged in through the front end, it assumes the user is whatever user is logged into your VM, in this case Loris admin. Try creating a user in your database that called lorisadmin. This should fix it

**Comment by sruthymathew123** [comment]
@leothomas  @ridz1208 I tested this script with different cases and seems to be working for me with no issues. 
The only thing I notice is that when adding an instrument to a time point, I got a PHP notice like this.
lorisadminPHP Notice:  Undefined index: CommentID in /var/www/loris/php/libraries/NDB_BVL_Feedback.class.inc on line 685
Added the instrument (vineland_subject) to the battery of the timepoint (1005)

Apart from that, everything seems to be ok.

**Comment by ridz1208** [comment]
@sruthymathew123 The PHP notice is okai since its originating from outside the script, it will be fixed in  another release.

please add the passed tests label and do the primary review

**Pull request merged**